x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/util/general.ts

```diff
@@ -75,7 +75,7 @@ export function getRichDetectors(
     }
 
     return {
-      agg: newJobCapsService.getAggById(d.function),
+      agg: d.function ? newJobCapsService.getAggById(d.function) : null,
```

Comment:
nit, function will always be defined.
```suggestion
      agg: newJobCapsService.getAggById(d.function!),
```